From a34167138fc1c61e228032a13ad3eb0b7b4f47b1 Mon Sep 17 00:00:00 2001
From: Muhammad Aksar <muhammad.aksar@intel.com>
Date: Wed, 20 Nov 2019 14:38:15 +0530
Subject: [PATCH] Camera: Fix crash while disconnecting USB Camera

unnecessarily sending error notification when
disconnecting the USB camera and hence it causes
camera service crash. Then need to kill the service
or restart the system forcefully. This patch would
be avoided this situation.

This patch will not effect MIPI-CSI Camera(s) or
other normal Camera use cases.

Tracked-On: OAM-88490
Signed-off-by: Muhammad Aksar <muhammad.aksar@intel.com>
---
 services/camera/libcameraservice/CameraService.cpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 3e62102..ec83193 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -358,13 +358,6 @@ void CameraService::onDeviceStatusChanged(const String8& id,
         if (clientToDisconnect.get() != nullptr) {
             ALOGI("%s: Client for camera ID %s evicted due to device status change from HAL",
                     __FUNCTION__, id.string());
-            // Notify the client of disconnection
-            clientToDisconnect->notifyError(
-                    hardware::camera2::ICameraDeviceCallbacks::ERROR_CAMERA_DISCONNECTED,
-                    CaptureResultExtras{});
-            // Ensure not in binder RPC so client disconnect PID checks work correctly
-            LOG_ALWAYS_FATAL_IF(CameraThreadState::getCallingPid() != getpid(),
-                    "onDeviceStatusChanged must be called from the camera service process!");
             clientToDisconnect->disconnect();
         }
 
-- 
2.7.4

